### ISSUE TITLE
@nrwl/js does not copy assets in watch mode

### ISSUE THREAD

**Issue by leanderpaul** [nrwl/nx#15105]
### Current Behavior

When watch option is passed to @nrwl/js package, the assets are not copied to the destination during the startup

### Expected Behavior

The assets should be copied to the destination given in the assets option

### GitHub Repo

https://github.com/leanderpaul/copy-assets-bug

### Steps to Reproduce

1. Run the code with the command `npx nx run server:serve` to get the error that file cannot be found
2. When the command `npx nx run server:serve` is used after setting the watch option to` false` in project.json, the code works.

### Nx Report

```shell
>  NX   Report complete - copy this into the issue template

   Node : 16.18.1
   OS   : linux x64
   npm  : 9.5.0

   nx                      : 15.7.2
   @nrwl/jest              : 15.7.2
   @nrwl/linter            : 15.7.2
   @nrwl/workspace         : 15.7.2
   @nrwl/cli               : 15.7.2
   @nrwl/devkit            : 15.7.2
   @nrwl/esbuild           : 15.7.2
   @nrwl/eslint-plugin-nx  : 15.7.2
   @nrwl/express           : 15.7.2
   @nrwl/js                : 15.7.2
   @nrwl/node              : 15.7.2
   @nrwl/tao               : 15.7.2
   @nrwl/webpack           : 15.7.2
   typescript              : 4.8.4
```

### Failure Logs

```shell
> nx run server:serve

[ watch ] build succeeded, watching for changes...
Debugger listening on ws://localhost:9229/e571c1e5-a19a-455b-9ded-8fb751e9c3c7
For help, see: https://nodejs.org/en/docs/inspector
Error: ENOENT: no such file or directory, open '<dir>/copy-assets-bug/dist/packages/server/assets/data.json'
    at Object.openSync (node:fs:590:3)
    at Object.readFileSync (node:fs:458:35)
    at Object.<anonymous> (<dir>/copy-assets-bug/packages/server/src/main.ts:8:20)
    at Module._compile (node:internal/modules/cjs/loader:1155:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1209:10)
    at Module.load (node:internal/modules/cjs/loader:1033:32)
    at Function.Module._load (node:internal/modules/cjs/loader:868:12)
    at Function.Module._load (<dir>/copy-assets-bug/packages/js/src/executors/node/node-with-require-overrides.ts:16:27)
    at Module.require (node:internal/modules/cjs/loader:1057:19)
    at require (node:internal/modules/cjs/helpers:103:18)
```

### Additional Information

 The error is because the package `@nrwl/js` is used internally to copy the assets, and when watch is set to `true`, the assets are not copied and only the changes that are made afterwards are handled.

When watch is set to `true`, the following function is called
https://github.com/nrwl/nx/blob/71fd015f3da26790ad265134dc3b345db1007576/packages/js/src/utils/assets/copy-assets-handler.ts#L141

and when the watch function is set to `false`, this function is called
https://github.com/nrwl/nx/blob/71fd015f3da26790ad265134dc3b345db1007576/packages/js/src/utils/assets/copy-assets-handler.ts#L104

so the solution would be to call the function `processAllAssetsOnce()` in the function `watchAndProcessOnAssetChange()` before the file watcher is registered

**Comment by j1mmie** [comment]
I am running into this issue as well. Thank you for the detailed detective work @leanderpaul, I'm sure that'll help resolve this faster.

To the nx team - any information on prioritization / timeline for this issue, or a recommended workaround, would be greatly appreciated.

**Comment by nartc** [comment]
This is fixed in #15230
